### PR TITLE
g203-led: init at 20210508-git

### PR DESCRIPTION
**Pull request by r-burns**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/g203-led/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv
+, fetchFromGitHub
+, python3
+}:
+stdenv.mkDerivation rec {
+  pname = "g203-led";
+  version = "20210508-git";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/g203-led/default.nix:7]:
```suggestion
  version = "unstable-2021-05-08";
```

File: pkgs/tools/misc/g203-led/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  meta = with lib; {
+    description = "Logitech G203 Prodigy / G203 LightSync Mouse LED control for Linux";
+    longDescription = ''
+      Allows you to control the LED lighting of your G203 Prodigy
+      or G203 LightSync Mouse programmatically.
+      Inspired by and based on g810-led.
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ r-burns ];
+    inherit (src.meta) homepage;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/g203-led/default.nix:37]:
```suggestion
    homepage = "https://github.com/smasty/g203-led";
```

File: pkgs/tools/misc/g203-led/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  postPatch = ''
+    patchShebangs g203-led.py
+  '';
+  doBuild = false;
+  installPhase = ''
+    runHook preInstall
+    install -D g203-led.py $out/bin/g203-led
+    runHook postInstall
+  '';
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/g203-led/default.nix:28]:
```suggestion
  };

  buildInputs = [
    (python3.withPackages (ps: with ps; [
      pyusb
    ]))
  ];

  postPatch = ''
    patchShebangs g203-led.py
  '';

  doBuild = false;

  installPhase = ''
    runHook preInstall
    install -D g203-led.py $out/bin/g203-led
    runHook postInstall
  '';

  meta = with lib; {
```

**Comment by r-burns** [comment]
Changes addressed.

File: pkgs/tools/misc/g203-led/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    (python3.withPackages (ps: with ps; [
+      pyusb
+    ]))
+  ];
+
+  postPatch = ''
+    patchShebangs g203-led.py
+  '';
+
+  doBuild = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/g203-led/default.nix:25]:
```suggestion
  dontBuild = true;
```

Comment (r-burns) [reply pkgs/tools/misc/g203-led/default.nix]: Good catch, I just removed it since true is the default

**Pull request merged**